docs/design/colocation/time-based-qos-policy.md

```diff
@@ -0,0 +1,213 @@
+If it is an `Active` event, it will append the policy to the currently tracked `activePolicies` and determine whether there are matching pods that need to be updated to the `targetQoSLevel`. 
+If it is an `Expired` event, it will remove the policy from the `activePolicies` and revert the matching pods to the original QoS level.
+`TimeBasedQoSHandle` is also responsible for processing `PodEvent` events generated by `PodProbe`. When a new pod is created on the node or the pod on the node is updated,
+`TimeBasedQoSHandle` traverses `activePolicies` to see if there is a matching policy. If there is a matching policy, the pod is updated to the `targetQoSLevel`.
+
+
+### Sequence Diagrams
+
+#### Policy Lifecycle Sequence
+
+![policylifecycle](images/policy-lifecycle.png)
+The above sequence diagram illustrates the lifecycle of a time based QoS policy. When a new policy is added or an existing policy is modified, 
+`TimeBasedQoSProbe` creates or updates a worker for that policy. The worker periodically checks if the current time falls within the policy's active period, 
+and sends appropriate events to the `TimeBasedQoSHandler`. Then `TimeBasedQoSHandler` processes these events to apply or revert QoS changes on matching pods.
+
+### Pod Annotation Management
+
+The system uses pod annotations to track QoS state changes:
+
+- `volcano.sh/qos-level`: Current QoS level assigned to the pod
+- `volcano.sh/original-qos-level`: This is a new annotation that needs to be added. It is used to persist the original qos level of the pod. 
+If the policy expires, the Handler will restore the pod's qos based on the original qos level when policy expires
+
+### Configuration Validation
+
+The system validates all policy configuration fields during configuration updates:
+
+- **Time Format Validation**: `StartTime` and `EndTime` must follow "HH:MM" format
+- **Time Zone Validation**: `TimeZone` must be a valid timezone identifier
+- **Selector Validation**: Label selectors must be valid Kubernetes label selectors
+- **Required Field Validation**: Critical fields(`TargetQoSLevel`, `StartTime`, `EndTime`, `Selector`) must be present and non-empty
+
+## Example
+
+### Configuration Example
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: volcano-agent-config
+  namespace: volcano-system
+data:
+  config.yaml: |
+    globalConfig:
+      timeBasedQoSPolicies:
+        - name: "online-qos-at-night"
+          enable: true
+          startTime: "22:00"
+          endTime: "08:00"
+          timeZone: "Asia/Shanghai"
+          selector:
+            matchLabels:
+              workload-type: "online"
+          targetQoSLevel: -1
+          checkInterval: "30s"
+        - name: "offline-qos-during-day"
+          enable: true
+          startTime: "08:00"
+          endTime: "22:00"
+          timeZone: "Asia/Shanghai"
+          selector:
+            matchLabels:
```

Comment:
So user needs to set lable for different workloads type when using this feature?